test/acceptance/tests/basic/basic_test.go

```diff
@@ -82,6 +83,49 @@ func TestBasic(t *testing.T) {
 				consulServerTaskARN = tasks.TaskARNs[0]
 			})
 
+			t.Cleanup(func() {
```

Comment:
It feels a bit weird to run this test inside of `t.Cleanup`. Maybe there's prior art here but I thought we usually use Cleanup to clean up things and it's relatively small.

What if we had a separate test for this functionality?

Reply:
Let me see if I can move it out cleanup. 

There is a cost to adding a new test just for this because it'll add quite a bit of execution time as the setup is the same. Ideally, these assertions are just part of the end-to-end service mesh workflow.